Sources/MapboxMaps/Location/LocationProviderDelegate.swift

```diff
@@ -40,3 +40,7 @@ internal class EmptyLocationProviderDelegate: LocationProviderDelegate {
     func locationProvider(_ provider: LocationProvider, didUpdateLocations locations: [CLLocation]) {}
     func locationProviderDidChangeAuthorization(_ provider: LocationProvider) {}
 }
+
+internal protocol CalibratingLocationProviderDelegate: LocationProviderDelegate {
+    func locationProviderShouldDisplayHeadingCalibration(_ locationProvider: LocationProvider) -> Bool
```

Comment:
as discussed, we should have this as part of public `LocationProviderDelegate` so custom `LocationProvider` can have one possible channel of communicating back to LocationManager to consult whether or not should calibration alert be displayed.

Reply:
Adding another method to a public protocol would be a breaking change and we need to verify that methods with default implementation are dispatched correctly before using those.